Support for global RPGLINT configuration path

francois-rpg opened the pull request:
### Changes

This PR adds support for specifying a global RPGLINT config path using the new setting rpgle.globalLintConfigPath.
If checked, linter configs for libraries will be based on IFS's /etc/vscode/rpglint.json.
This helps enforce company-wide lint standards.
Exclude libraries also added, to enable user to add libraries they don't want linted, e.g. browsing prod libraries' source.

### Checklist

* [Y] have tested my change
* [ ] updated relevant documentation
* [ ] Remove any/all `console.log`s I added
* [ ] eslint is not complaining
* [ ] have added myself to the contributors' list in the README
* [ ] **for feature PRs**: PR only includes one feature enhancement.

### This is my first pull request to anything on github, ever. Please let me know if and where I went wrong so I contribute cleanly in the future.

francois-rpg committed:
Update lockfile version

francois-rpg committed:
Merge pull request #1 from RPGLE-Logic

Add global lint config path support

francois-rpg committed:
Create package-extension.yml

francois-rpg committed:
Update package-extension.yml

francois-rpg committed:
Create tsconfig.json

francois-rpg committed:
Update tsconfig.json

francois-rpg committed:
Update package-extension.yml

francois-rpg committed:
Update tsconfig.json

francois-rpg committed:
Update tsconfig.json

francois-rpg committed:
Update tsconfig.json

francois-rpg committed:
Update tsconfig.json

francois-rpg committed:
Upload vsix file

Compiled in github codespaces

francois-rpg committed:
feat: support global lint config

francois-rpg committed:
Merge pull request #2 from RPGLE-Logic

Add global lint config option

francois-rpg committed:
Add files via upload

francois-rpg committed:
Update linter.ts

francois-rpg committed:
Update package.json

francois-rpg committed:
Update linter.ts

francois-rpg committed:
Add files via upload

worksofliam reviewed (changes requested):
Hey, this is a great start! I am going to need some files removed from this PR:

* `package-extension.yml` - please don't add any new actions to this PR. Perhaps in a separate one if this is something you want to see.
* Please remove all `.vsix` files. We don't want to check them into the repo.

Thanks! Then I will go in for a second review 😄

francois-rpg committed:
Unneeded files removed

RPGLE-Logic commented:
Requested changes completed.
Thank you for the feedback.

worksofliam reviewed (changes requested):
Sorry this has token so long to get to.

1. Can you please fix the formatting in your new changes? They don't match the indentation of the rest of the file.
2. Can you fix the conflicts

Thanks!

francois-rpg committed:
Update and additions

-Correcting indentation.
-Adding ignore directories config

francois-rpg committed:
Undo unintended changes

francois-rpg committed:
Merge branch 'main' into pr/435

francois-rpg committed:
Applied suggested changes

Changed global path to use IFS.
Brought in line with issue PR fixing lint creation.

francois-rpg committed:
Merge branch 'main' into main
